### PR TITLE
refactor(query): stage/tables/databases DDL statements defaults to use new planner

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Related #6644

### CODE REVIEW

**Comment by Xuanwo** [comment]
```
2022-07-18T14:14:23.942120Z ERROR databend_query::servers::mysql::writers::query_result_writer: OnQuery Error: Code: 3002, displayText = permission denied (object error: (op: list, path: admin/data/, source: status code: 403, headers: {"x-amz-bucket-region": "us-east-1", "x-amz-request-id": "QEBSV446M20CVZ3F", "x-amz-id-2": "Ki/3AjozX4xCJ/TuW37A5QthzypUJHTKE3ZPgcghuOZ5HH4xso9yAs7Zyl+yecqluyzPFKytWEo=", "content-type": "application/xml", "transfer-encoding": "chunked", "date": "Mon, 18 Jul 2022 14:14:23 GMT", "server": "AmazonS3"}, body: "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>InvalidAccessKeyId</Code><Message>The AWS Access Key Id you provided does not exist in our records.</Message><AWSAccessKeyId>minioadmin</AWSAccessKeyId><RequestId>QEBSV446M20CVZ3F</RequestId><HostId>Ki/3AjozX4xCJ/TuW37A5QthzypUJHTKE3ZPgcghuOZ5HH4xso9yAs7Zyl+yecqluyzPFKytWEo=</HostId></Error>")).
```

After migrating to the new planner, the default endpoint for COPY will become `https://s3.amazonaws.com` instead of the query's storage backend. To fix stateful tests, we need to add `endpoint_uil = 'http://127.0.0.1:9900'` in `connection`.

For example:

```sql
copy into ontime200 from 's3://testbucket/admin/data/ontime_200.csv' connection=(aws_key_id='minioadmin' aws_secret_key='minioadmin' endpoint_url='${STORAGE_S3_ENDPOINT_URL}') FILE_FORMAT = (type = 'CSV' field_delimiter = ','  record_delimiter = '\n' skip_header = 1)
```

**Comment by sundy-li** [comment]
> ```
> 2022-07-18T14:14:23.942120Z ERROR databend_query::servers::mysql::writers::query_result_writer: OnQuery Error: Code: 3002, displayText = permission denied (object error: (op: list, path: admin/data/, source: status code: 403, headers: {"x-amz-bucket-region": "us-east-1", "x-amz-request-id": "QEBSV446M20CVZ3F", "x-amz-id-2": "Ki/3AjozX4xCJ/TuW37A5QthzypUJHTKE3ZPgcghuOZ5HH4xso9yAs7Zyl+yecqluyzPFKytWEo=", "content-type": "application/xml", "transfer-encoding": "chunked", "date": "Mon, 18 Jul 2022 14:14:23 GMT", "server": "AmazonS3"}, body: "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>InvalidAccessKeyId</Code><Message>The AWS Access Key Id you provided does not exist in our records.</Message><AWSAccessKeyId>minioadmin</AWSAccessKeyId><RequestId>QEBSV446M20CVZ3F</RequestId><HostId>Ki/3AjozX4xCJ/TuW37A5QthzypUJHTKE3ZPgcghuOZ5HH4xso9yAs7Zyl+yecqluyzPFKytWEo=</HostId></Error>")).
> ```
> 
> After migrating to the new planner, the default endpoint for COPY will become `https://s3.amazonaws.com` instead of the query's storage backend. To fix stateful tests, we need to add `endpoint_uil = 'http://127.0.0.1:9900'` in `connection`.
> 
> For example:
> 
> ```sql
> copy into ontime200 from 's3://testbucket/admin/data/ontime_200.csv' connection=(aws_key_id='minioadmin' aws_secret_key='minioadmin' endpoint_url='${STORAGE_S3_ENDPOINT_URL}') FILE_FORMAT = (type = 'CSV' field_delimiter = ','  record_delimiter = '\n' skip_header = 1)
> ```

Oh, thanks, I was stuck into this failed ci for hours.

**Review by Xuanwo — APPROVED** [review]
Huge steps!

File: common/ast/src/parser/statement.rs [hunk 1]
```diff
@@ -1177,16 +1199,23 @@ pub fn show_limit(i: Input) -> IResult<ShowLimit> {
...
-        map(
-            rule! {
-                COMMENT ~ ^"=" ~ #literal_string
-            },
-            |(_, _, comment)| TableOption::Comment(comment),
-        ),
-    ))(i)
+pub fn table_option(i: Input) -> IResult<BTreeMap<String, String>> {
+    let opt_key = map_res(ident, |ident| {
+        if ident.quote.is_none() && ident.name != "AS" {
```
Comment (andylokandy) [inline common/ast/src/parser/statement.rs:1204]: `AS` should be a reserved keyword and is impossible to be an identifier unless it's parsed using `ident_after_as`.

File: common/ast/tests/it/testdata/statement.txt [hunk 2]
```diff
@@ -234,7 +234,7 @@ Some(
 ---------- Input ----------
 create table if not exists a.b (c integer not null default 1, b varchar);
 ---------- Output ---------
-CREATE TABLE IF NOT EXISTS a.b (c Int32 NOT NULL DEFAULT 1, b STRING NOT NULL)
+CREATE TABLE IF NOT EXISTS a.b (c Int32 DEFAULT 1, b STRING)
```
[the remark this answers is not in the document]
Comment (sundy-li) [reply common/ast/tests/it/testdata/statement.txt:237]: Do you suggest that we should not omit the `NOT NULL` ?

**Comment by sundy-li** [comment]
`test_stateless_cluster_macos` is flaky, it succeeds after serval retries...

**Review by andylokandy — APPROVED** [review]

**Pull request merged**